Analytical_Engine/Query/IsAligned.cs

```diff
@@ -44,24 +44,23 @@ public static partial class Query
         /**** Public Methods                            ****/
         /***************************************************/
 
-        [Description("Determines whether a panel is aligned.")]
-        [Input("panel", "The IPanel to check if a panel is aligned.")]
-        [Input("PlaneDirection", "A user defined plane direction .")]
-        [Output("bool", "True for panels for panels is aligned.")]
-        public static bool IsAligned<TEdge, TOpening>(this IPanel<TEdge, TOpening> panel,Plane PlaneDirection)
+        [Description("Checks if the Panel is aligned to a defined Plane.")]
+        [Input("panel", "The IPanel to check alignment for.")]
+        [Input("plane", "The Plane that the Panel is aligned to check against.")]
+        [Output("bool", "True for panels is aligned.")]
+        public static bool IsAligned<TEdge, TOpening>(this IPanel<TEdge, TOpening> panel, Plane plane)
             where TEdge : IEdge
             where TOpening : IOpening<TEdge>
         {
             PolyCurve polycurve = ExternalPolyCurve(panel);
 
             if (!polycurve.IsPlanar())
                 return false;
-            Vector tangent1 = polycurve.TangentAtParameter(1);
-            if (tangent1.IsInPlane(PlaneDirection, Tolerance.Distance) == false)
-                return false;
+            Vector vector = plane.Normal;
+            Vector tangent = polycurve.Normal();
```

Comment:
I think there's a benefit from a code readability and maintainability reason - you're right that your suggestion is a nice one-line solution, but from a debugging and maintenance perspective it can be better to take the variables out. I would thus advise against your suggestions @peterjamesnugent in this instance.

Reply:
@FraserGreenroyd good points, @StephennipBH can we rename to:
`planeNormal` and `polycurveNormal`, I think the two current names are a hangover from previous code.